### ISSUE TITLE
Retry logs full request exposing secrets

### ISSUE THREAD

**Issue by channingwalton** [http4s/http4s#6597]
When Retry fails the request is logged in the error in its entirety, including query parameters, which can expose sensitive information, such as API secrets or login credentials, in logs.

[This](https://github.com/http4s/http4s/blob/c383f61af6ce13d92245889acc9445b07c91a7c9/client/shared/src/main/scala/org/http4s/client/middleware/Retry.scala#L126) is the line of code that does it I believe.

Perhaps Retry should require a Show[Request] or something similar to [redactHeaderWhen](https://github.com/http4s/http4s/blob/c383f61af6ce13d92245889acc9445b07c91a7c9/client/shared/src/main/scala/org/http4s/client/middleware/Retry.scala#L52)?

**Comment by danicheg** [comment]
I can be wrong, but `Retry` contains redacting of the headers already. Do you pointing that we should expose the ability to redact the URI of the request?

**Comment by channingwalton** [comment]
Yes, redact the URI by requiring a Show[Uri] or something like it.

**Comment by armanbilge** [comment]
@channingwalton fwiw that is a well-known risk broader than http4s and generally seems to be an anti-pattern.

https://owasp.org/www-community/vulnerabilities/Information_exposure_through_query_strings_in_url

**Comment by channingwalton** [comment]
> @channingwalton fwiw including sensitive info in a URI is a well-known risk broader than http4s and generally seems to be an anti-pattern.

Yes, I am working with a 3rd party API so forced to do it.

**Comment by channingwalton** [comment]
@ChristopherDavenport said on Discord:

> LogRetries should just be false and have logging operate through your Middleware.
> It was left for old times and bincompat, but automatic logging is just dangerous.

**Comment by armanbilge** [comment]
I agree that the `logRetries = true` default is unfortunate. We can take a PR to change it on the `main` branch.

It sounds like in your case you should use `false` since you do not control the offending API.

**Comment by channingwalton** [comment]
The are two options I believe:

1. BlazeClient sets `logRetries = false`
2. Retry defaults to `logRetries = false` 

Which would be preferable? I am thinking `2`.

**Comment by armanbilge** [comment]
At least in this repo, definitely (2). I'm not sure what's appropriate for blaze, you may want to bring it up in https://github.com/http4s/blaze.

**Comment by channingwalton** [comment]
Ok. I'm a bit confused about where things are but I'll start with blaze.
